popup/popup.js

```diff
@@ -1,5 +1,115 @@
-document.addEventListener('DOMContentLoaded', function() {
-    chrome.storage.local.get(['productData'], function(result) {
-        document.getElementById('data').innerText = JSON.stringify
+const proxyUrl = 'https://cors-anywhere.herokuapp.com/';
+const targetUrl = 'https://www.bing.com/api/shopping/v1/item/search?appid=67220BD2169C2EA709984467C21494086DF8CA85&features=persnlcashback&sf=cashback1';
+const url = proxyUrl + targetUrl;
+
+chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+    let tabUrl = new URL(tabs[0].url);
+    let domain = tabUrl.hostname;
+    let path = tabUrl.pathname;
+
+    if (domain.startsWith('www.')) {
+        domain = domain.substring(4);
+    }
+    
+    console.log(domain);
+    console.log(path);
+    const body = {
+        "AgeGroup": 0,  
+        "IsAADSignedIn": true,
+        "IsBingSignedInUser": false,
+        "IsMSASignedIn": false,
+        "IsOffTheRecord": false,
+        "IsPersonalizationDataConsentChanged": false,
+        "IsPersonalizationDataConsentEnabled": false,
+        "IsSSOEnabled": false,
+        "LdClickData": "",
+        "MsClickId": "",
+        "ReturnedToCashbackActivatedDomain": false,
+        "UClickData": "",
+        "item": {
+            "seller": {
+                "IsEdgeRebatesFlow": false,
+                "domain": domain,
+                "path": path
+            }
+        },
+        "sourceTypes": [
+            "deals"
+        ]
+    };
+    
+    async function postData(url = '', data = {}) {
+        const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json', 
+                'X-Requested-With': 'XMLHttpRequest'
+            },
+            body: JSON.stringify(data)
+        });
+        return response.json(); // parses JSON response into native JavaScript objects
+    }
+    
+    postData(url, body)
+        .then(data => {
+            console.log(data); // JSON data parsed by `response.json()` call
+            const allCheckoutPages = data.deals.retailerData.allCheckoutPages;
+            let matchingPage = allCheckoutPages.find(page => page.checkoutPageUrl === path);
+    
+            if (!matchingPage) {
+                matchingPage = allCheckoutPages.find(page => path.startsWith(page.checkoutPageUrl));
```

Comment:
This follows up if there is a incorrect matching, but why at all give the control here?
Suggestion would be to use else, as a fallback if a correct match fails, in this scenario we are performing a not so robust match and asking for a else to deal with the aftermath. It would be better to use if more robustly.